en/rest/hooks.mdown

```diff
@@ -0,0 +1,670 @@
+can be one of `beforeSave`, `afterSave`, `beforeDelete`, `afterDelete`.
+
+The output may look like this:
+```json
+{
+  "results": [
+    { "className": "Scores", "triggerName": "beforeSave" },
+    {
+      "className": "Scores",
+      "triggerName": "beforeSave",
+      "url": "https://api.example.com/Scores/beforeSave"
+    }
+  ]
+}
+```
+
+The output is a json object with one key: "results" whose value is a list of all
+cloud code triggers with given name for a given class.
+
+If only one of cloud code trigger or cloud code trigger webhook with the given name exists,
+then only that one value is present in the list.
+
+For example,
+```bash
+curl -X GET \
+  -H "X-Parse-Application-Id: ${APPLICATION_ID}" \
+  -H "X-Parse-Master-Key: ${MASTER_KEY}" \
+  -H "Content-Type: application/json" \
+  https://api.parse.com/1/hooks/triggers/Game/afterSave
+```
+```python
+import json,httplib
+connection = httplib.HTTPSConnection('api.parse.com', 443)
+connection.connect()
+connection.request('GET', '/1/hooks/triggers/Game/afterSave', '', {
+       "X-Parse-Application-Id": "${APPLICATION_ID}",
+       "X-Parse-Master-Key": "${MASTER_KEY}",
+       "Content-Type": "application/json"
+     })
+result = json.loads(connection.getresponse().read())
+print result
+```
+
+The output may look like this:
+```json
+{
+  "results": [
+    {
+      "className": "Game",
+      "triggerName": "afterSave",
+      "url": "https://api.example.com/Game/afterSave"
+    }
+  ]
+}
+```
+
+Note that POST, PUT and DELETE only work on function or trigger webhooks.
+To create regular cloud code functions or triggers you can modify your cloude code js files
+and perform a `parse deploy` the usual way.
+
+## Create function webhook
+To create a new cloud code function webhook post to `/1/hooks/functions` with payload in the format
+```bash
+{"functionName" : x, "url" : y}
+```
+
+Post example,
+
+```bash
+curl -X POST \
+  -H "X-Parse-Application-Id: ${APPLICATION_ID}" \
+  -H "X-Parse-Master-Key: ${MASTER_KEY}" \
+  -H "Content-Type: application/json" \
+  -d '{"functionName":"baz","url":"https://api.example.com/baz"}' \
+  https://api.parse.com/1/hooks/functions
+```
+```python
+import json,httplib
+connection = httplib.HTTPSConnection('api.parse.com', 443)
+connection.connect()
+connection.request('POST', '/1/hooks/functions', json.dumps(
+       {"functionName":"baz","url":"https://api.example.com/baz"}
+     ), {
+       "X-Parse-Application-Id": "${APPLICATION_ID}",
+       "X-Parse-Master-Key": "${MASTER_KEY}",
+       "Content-Type": "application/json"
+     })
+result = json.loads(connection.getresponse().read())
+print result
+```
+
+The output may look like this:
+```json
+{"functionName": "baz", "url": "https://api.example.com/baz"}'
+```
+
+It returns the function name and url of the created webhook.
+
+If a cloud code function with the same name already exists then the result also contains a warning.
```

Comment:
When a warning is shown, it's unclear to me whether the newly-requested changes were actually committed.  Could we clarify this in the docs?